app/controllers/user_controller.rb

```diff
@@ -6,12 +6,17 @@ def index
   end
 
   def signup
+    error_messages = Validation.sign_up_validate(user_params)
+    if !error_messages.empty?
+      render json: { "errors": error_messages }, status: 400
+      return
+    end
     @new_user = User.new(user_params)
 
     if @new_user.valid?
       @new_user.save
-
-      render json: @new_user, status: 201
+      user = { email: @new_user.email, id: @new_user.id, first_name: @new_user.first_name, last_name: @new_user.last_name, user_type: @new_user.user_type }
+      render json: user, status: 201
```

Comment:
A couple of other ways to serialize model:

https://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json
https://api.rubyonrails.org/classes/ActiveModel/Serialization.html